### PR TITLE
Use unittest.TestCase.id() to put together jUnit XML output.

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says. This is particularly important if using `unittest.TestCase.subTest()` or otherwise a generic name is all you get (see [an example](https://ci.ros2.org/view/nightly/job/nightly_linux_debug/1602/testReport/junit/ros2topic.ros2topic.test/test_echo_pub/test_echo_pub/)).

### CODE REVIEW

**Comment by hidmic** [comment]
Full CI to make sure nothing breaks:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11372/)

File: launch_testing/launch_testing/junitxml.py [hunk 1]
```diff
@@ -83,9 +83,10 @@ def unittestCaseToXml(test_result, test_case):
     class needs to be a launch_testing.TestResult class
     """
     case_xml = ET.Element('testcase')
-    full_classname = type(test_case).__module__ + '.' + type(test_case).__name__
+    full_methodname, _, qualifiers = test_case.id().partition(' ')
```
Comment (ivanpauno) [inline launch_testing/launch_testing/junitxml.py:86]: can you give an example of how a full method name with qualifiers look?

File: launch_testing/test/launch_testing/test_xml_output.py [hunk 2]
```diff
@@ -86,6 +86,8 @@ def unit_test_result_factory(self, test_case_list):
         class TestHost(unittest.TestCase):
             pass
 
+        TestHost.__qualname__ = 'TestHost'  # as if it was defined in global scope
```
Comment (ivanpauno) [inline launch_testing/test/launch_testing/test_xml_output.py:89]: why is this needed?

File: launch_testing/launch_testing/junitxml.py (hunk 1 above)
Comment (hidmic) [reply launch_testing/launch_testing/junitxml.py:86]: Qualifiers apply to the test case, not the method. When using the [`subTest` feature](https://docs.python.org/3/library/unittest.html#unittest.TestCase.subTest), `unittest` appends message and parameters that I need to account for.

File: launch_testing/test/launch_testing/test_xml_output.py (hunk 2 above)
Comment (hidmic) [reply launch_testing/test/launch_testing/test_xml_output.py:89]: With the change in how we generate jUnit XML, tests that make use of this local class would have to be changed to account for its real qualified class name i.e. it's not `test_xml_output.TestHost`, more like `test_xml_output.unit_test_result_factory.<locals>.TestHost`. And I didn't think it was worth cluttering tests with that detail, therefore this line.

File: launch_testing/launch_testing/junitxml.py (hunk 1 above)
Comment (ivanpauno) [reply launch_testing/launch_testing/junitxml.py:86]: thanks for clarifying, sounds good

**Comment by hidmic** [comment]
Alright, full Linux CI's green. This is trivial Python code, so I don't think it's worth running full CI on other platforms.

**Comment by dirk-thomas** [comment]
Maybe I am missing something but how does the what improvement look like for the example above? The latest case with this patch seems to have the exact same output: https://ci.ros2.org/view/nightly/job/nightly_linux_debug/1606/testReport/ros2topic.ros2topic.test/test_echo_pub/test_echo_pub/

**Comment by hidmic** [comment]
Uh, that's... unexpected.
